### PR TITLE
feat(groovy): Add more support for groovy syntax. Fix for groovy multiline comment highlighting.

### PR DESCRIPTION
**Pull request by run2cmd**

### CODE REVIEW

File: queries/groovy/highlights.scm [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
 (func
-    (identifier) @function)
+  (identifier) @function
+)
 
 (number) @number
 
 ((identifier) @boolean
-  (#any-of? @boolean "true" "false" "True" "False"))
+  (#any-of? @boolean "true" "false" "True" "False" "null"))
```
Comment (ObserverOfTime) [inline queries/groovy/highlights.scm:21]: "null" should be `@constant.builtin`.

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]
Don't change the bracket formatting.

**Comment by run2cmd** [comment]
> Don't change the bracket formatting.

reverted

File: queries/groovy/highlights.scm [hunk 2]
```diff
@@ -36,11 +44,28 @@
 ((identifier) @keyword
   (#any-of? @keyword
     "static"
+    "abstract"
+    "protected"
+    "private"
+    "public"
```
Comment (ObserverOfTime) [inline queries/groovy/highlights.scm:50]: These can be `@type.qualifier`.

Comment (run2cmd) [reply queries/groovy/highlights.scm:50]: Done.

File: queries/groovy/highlights.scm [hunk 3]
```diff
@@ -39,8 +47,29 @@
...
+    "interface"
     "new"))
 
+((identifier) @type.qualifier
+  (#any-of? @type.qualifier
+    "abstract"
+    "protected"
+    "private"
+    "public"
+))
```
Comment (ObserverOfTime) [inline queries/groovy/highlights.scm:64]: Formatting. :)
```suggestion
    "public"))
```

File: queries/groovy/highlights.scm [hunk 4]
```diff
@@ -20,10 +20,18 @@
...
 ((identifier) @type.definition
   (#any-of? @type.definition
     "string"
     "String"
+    "Map"
+    "Object"
+    "Boolean"
+    "Integer"
+    "List"
     "void"
```
Comment (ObserverOfTime) [inline queries/groovy/highlights.scm:35]: I just noticed this shouldn't be `@type.definition`.
I'd make these `@type` and the rest `@function.builtin`.

File: queries/groovy/highlights.scm (hunk 3 above)
Comment (run2cmd) [reply queries/groovy/highlights.scm]: fixed

File: queries/groovy/highlights.scm (hunk 4 above)
Comment (run2cmd) [reply queries/groovy/highlights.scm]: Changed

**Comment by amaanq** [comment]
For the future don't use master as your PR branch, it causes problems checking out/pushing to for others

**Review by amaanq — APPROVED** [review]

**Comment by amaanq** [comment]
@ObserverOfTime think you need to approve it

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**